src/emu/romentry.h

```diff
@@ -0,0 +1,153 @@
+#define     ROMREGION_ERASE         0x00002000
+
+#define ROMREGION_DATATYPEMASK      0x00004000          /* type of region (ROM versus disk) */
+#define     ROMREGION_DATATYPEROM   0x00000000
+#define     ROMREGION_DATATYPEDISK  0x00004000
+
+#define ROMREGION_ERASEVALMASK      0x00ff0000          /* value to erase the region to */
+#define     ROMREGION_ERASEVAL(x)   ((((x) & 0xff) << 16) | ROMREGION_ERASE)
+#define     ROMREGION_ERASE00       ROMREGION_ERASEVAL(0)
+#define     ROMREGION_ERASEFF       ROMREGION_ERASEVAL(0xff)
+
+
+	/* ----- per-ROM constants ----- */
+#define DISK_READONLYMASK           0x00000010          /* is the disk read-only? */
+#define     DISK_READWRITE          0x00000000
+#define     DISK_READONLY           0x00000010
+
+#define ROM_OPTIONALMASK            0x00000020          /* optional - won't hurt if it's not there */
+#define     ROM_REQUIRED            0x00000000
+#define     ROM_OPTIONAL            0x00000020
+
+#define ROM_REVERSEMASK             0x00000040          /* reverse the byte order within a group */
+#define     ROM_NOREVERSE           0x00000000
+#define     ROM_REVERSE             0x00000040
+
+#define ROM_INHERITFLAGSMASK        0x00000080          /* inherit all flags from previous definition */
+#define     ROM_INHERITFLAGS        0x00000080
+
+#define ROM_GROUPMASK               0x00000f00          /* load data in groups of this size + 1 */
+#define     ROM_GROUPSIZE(n)        ((((n) - 1) & 15) << 8)
+#define     ROM_GROUPBYTE           ROM_GROUPSIZE(1)
+#define     ROM_GROUPWORD           ROM_GROUPSIZE(2)
+#define     ROM_GROUPDWORD          ROM_GROUPSIZE(4)
+
+#define ROM_SKIPMASK                0x0000f000          /* skip this many bytes after each group */
+#define     ROM_SKIP(n)             (((n) & 15) << 12)
+#define     ROM_NOSKIP              ROM_SKIP(0)
+
+#define ROM_BITWIDTHMASK            0x000f0000          /* width of data in bits */
+#define     ROM_BITWIDTH(n)         (((n) & 15) << 16)
+#define     ROM_NIBBLE              ROM_BITWIDTH(4)
+#define     ROM_FULLBYTE            ROM_BITWIDTH(8)
+
+#define ROM_BITSHIFTMASK            0x00f00000          /* left-shift count for the bits */
+#define     ROM_BITSHIFT(n)         (((n) & 15) << 20)
+#define     ROM_NOSHIFT             ROM_BITSHIFT(0)
+#define     ROM_SHIFT_NIBBLE_LO     ROM_BITSHIFT(0)
+#define     ROM_SHIFT_NIBBLE_HI     ROM_BITSHIFT(4)
+
+#define ROM_BIOSFLAGSMASK           0xff000000          /* only loaded if value matches global bios value */
+#define     ROM_BIOS(n)             (((n) & 255) << 24)
+
+#define ROM_INHERITEDFLAGS          (ROM_GROUPMASK | ROM_SKIPMASK | ROM_REVERSEMASK | ROM_BITWIDTHMASK | ROM_BITSHIFTMASK | ROM_BIOSFLAGSMASK)
+
+
+
+//**************************************************************************
+//  TYPE DEFINITIONS
+//**************************************************************************
+
+// ======================> rom_entry
+
+class rom_entry
+{
+public:
```

Comment:
Don't forget to add the annotation necessary for the compiler to generate move construct/assign